include/inviwo/core/interaction/contextmenuaction.h

```diff
@@ -0,0 +1,105 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2025 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#pragma once
+
+#include <inviwo/core/common/inviwocoredefine.h>
+#include <inviwo/core/util/fmtutils.h>
+
+#include <flags/flags.h>
+#include <string_view>
+#include <iosfwd>
+#include <cstdint>
+#include <span>
+#include <string>
+#include <vector>
+#include <optional>
+#include <variant>
+
+namespace inviwo {
+
+struct ContextMenuAction;
+struct ContextMenuSeparator;
+struct ContextMenuSubmenu;
+using ContextMenuEntry = std::variant<ContextMenuAction, ContextMenuSubmenu, ContextMenuSeparator>;
+
+/**
+ * Datastructure for a custom menu action in a context menu. The id should be unique, for example
+ * <tt><getIdentifier()>.myAction</tt>. Otherwise it cannnot be guaranteed that the correct
+ * InteractionHandler/Processor handles the corresponding ContextMenuEvent. If a menu action is
+ * triggered, a ContextMenuEvent with the id will be propagated through the processor network.
+ *
+ * @see ContextMenuEvent ContextMenuSeparator ContextMenuSubmenu
+ */
+struct ContextMenuAction {
+    std::string label;
+    std::string id;
```

Comment:
Think we might have talked about adding a std::any to this also, to be able to pass some payload long. But I think we can add that in a later PR.

Reply:
Future PR sounds good. For now some basic functionality is sufficient I'd say.